### PR TITLE
Fetch updates from i.d.s by default

### PR DESCRIPTION
**Pull request by grahamc**
Our goal is to deliver a solid experience with Nix flakes, including an upgrade path that is safe. Occasionally, the upstream Nix project may introduce regressions for the common flake path. This is not desirable for our users, who depend on a consistent and stable flakes experience.

Additionally, the Nix project isn't directly responsible for delivering updates to users as that role is delegated to the Nixpkgs project.

Overall, this means upgrades are not consistently delivered to users.

This update directs future update requests to install.determinate.systems, which we will upgrade as part of our standard release process.

Our standard release process includes proactive testing: validating our installer and Nix's behavior across a wide variety of platforms and scenarios.

After an update passes our proactive validation, we do a phased rollout of reactive monitoring: the update is released to a small percentage of users on GitHub Actions. We monitor the failure rate of the installer and overall workflows to ensure the updated Nix isn't causing widespread failure we weren't able to identify ahead of time.

Only after a release passes both proactive and reactive validation, our macOS .pkg and nix-upgrade paths are bumped to the most recent release. This gives user the confidence they're looking for that the Nix release they're getting is safe.

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

**Comment by grahamc** [comment]
Closes #744

File: src/action/common/place_nix_configuration.rs [hunk 1]
```diff
@@ -129,6 +129,10 @@ impl PlaceNixConfiguration {
             "extra-nix-path".to_string(),
             "nixpkgs=flake:nixpkgs".to_string(),
         );
+        settings.insert(
+            "upgrade-nix-store-path-url".to_string(),
+            "https://install.determinate.systems/nix-upgrade/stable/Universal".to_string(),
```
Comment (Hoverbear) [inline src/action/common/place_nix_configuration.rs:134]: We discussed either directly pointing at, or linking to, https://github.com/DeterminateSystems/nix-upgrade

Comment (grahamc) [reply src/action/common/place_nix_configuration.rs:134]: All set, thanks! Once the install.determinate.system update rolls out we'll be serving this by default:

```
$ curl -L https://install.determinate.systems/nix-upgrade/tag/v2.19.2+1/Universal
# Mirrored from https://releases.nixos.org/nix/nix-2.19.2/fallback-paths.nix
# via https://github.com/DeterminateSystems/nix-upgrade

{
  x86_64-linux = "/nix/store/smfmnz0ylx80wkbqbjibj7zcw4q668xp-nix-2.19.2";
  i686-linux = "/nix/store/knp0akbpj2k0rf26fmysmxdysmayihax-nix-2.19.2";
  aarch64-linux = "/nix/store/761hq0abn07nrydrf6mls61bscx2vz2i-nix-2.19.2";
  x86_64-darwin = "/nix/store/zlqvxis1dfcfgmy5fza4hllg6h03vhpb-nix-2.19.2";
  aarch64-darwin = "/nix/store/53r8ay20mygy2sifn7j2p8wjqlx2kxik-nix-2.19.2";
}
```

Comment (grahamc) [reply src/action/common/place_nix_configuration.rs:134]: 👍 ready to roll :)

**Comment by fesplugas** [comment]
A couple days ago I upgraded my nix install and now I see a warning when running flakes

```
warning: unknown setting 'upgrade-nix-store-path-url'
```

Removing the setting from `/etc/nix/nix.conf` fixes the warning, but I wonder if there are any side effects.

**Comment by cole-h** [comment]
The only side effect is that you won't get the latest Nix version, but will be beholden to the Nixpkgs maintainers' decisions on which version of Nix to provide in https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/installer/tools/nix-fallback-paths.nix.

If you say you "upgraded" your Nix install but are now seeing that, it seems more likely that you downgraded it (whether on purpose or on accident)? That option was added in Nix 2.19.0 (https://github.com/NixOS/nix/pull/9333), and this PR first appeared in `nix-installer` version 0.16.0 (https://github.com/DeterminateSystems/nix-installer/releases/tag/v0.16.0), which installed Nix 2.19.2 (and any later versions of the installer should install 2.19.2 or newer versions of Nix).

What version of Nix are you using? `nix --version`

**Pull request merged**